lib/ddtrace/method_tracing.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Datadog
+  # \MethodTracing adds simple helpers for adding
+  # tracing to your methods.
+  #
+  # = Example
+  #
+  #   extend Datadog::MethodTracing
+  #   trace_methods self, :method1, :method2
+  module MethodTracing
+    # Wrap *methods* on *klass* in a trace block
+    def trace_methods(klass, *methods)
+      methods.each { |m| trace_method(klass, m) }
+    end
+
+    def trace_method(klass, method)
+      metric = "#{class_name(klass)}.#{method}"
+      m = Module.new do
+        define_method(method) do |*args, &block|
+          Datadog.tracer.trace(metric) { super(*args, &block) }
+        end
+      end
+      klass.prepend(m)
+    end
+
+    def class_name(klass)
```

Comment:
Because `MethodTracing` will be included in the user's class, we should reduce its footprint as much as possible in order to not conflict the original class' methods.

I think it would make sense to extract `class_name`, `demodulize` and `underscore` into their own module/class and call them directly from `#trace_method`.